A solution for the third hw (timer)

Renalime opened the pull request:
This is a solution for the third homework that requires usage of the Linux Kernel Time Management API.
All the logs are in README.md file.

Renalime committed:
timer: Add a userspace program for absolute time

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
timer: Add a module file with init and exit

In the module, the last_read variable is initialized during module
initialization.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
timer: Add a class and interval attribute

During init the mytimer class and the interval attribute are created.
All the intervals are passed in nanoseconds.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
timer: Add 'time' attribute to the mytimer class

The 'time' attribute gives the output in seconds.nanoseconds format.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
timer: Add dummy load attr and timer setup

Added load attribute that return always 0%, and a timer setup with
a function callback.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
Merge branch 'Andrii.Revva' into timer

Renalime committed:
timer: Timer callback registers itself

Made timer to register itself in callback in order to check cpu load
periodically.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

yekovalyov committed:
time management: homework

Signed-off-by: Yevgen Kovalyov <yevgen.kovalyov@globallogic.com>

yekovalyov committed:
memory management: homework

Signed-off-by: Yevgen Kovalyov <yevgen.kovalyov@globallogic.com>

Renalime committed:
timer: Add proper cpu load accounting

The timer callback each time calculates cpu_load, based on the previous
load information and registers itself each seccond.
CPU load info is passed to kernel buffer.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
Merge branch 'master' into Andrii.Revva

Renalime committed:
Merge branch 'Andrii.Revva' into timer

Renalime committed:
timer: Add solution description to README

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

yekovalyov commented:
Please check your code using checkpatch.pl. It is code style checker from Linux repository folder scripts.

Renalime committed:
timer: Improve code to comply with checkpatch

All the errors and some warnings reported by checkpatch were fixed, except
those that report: "struct should normally be const".

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime commented:
@yekovalyov Done with checkpatch. Details about what was fixed are in the last commit message.

yekovalyov commented:
Excellent work!
